prov/util/src/util_coll.c

```diff
@@ -856,28 +851,28 @@ int ofi_av_set(struct fid_av *av, struct fi_av_set_attr *attr,
 
 ssize_t ofi_ep_barrier(struct fid_ep *ep, fi_addr_t coll_addr, void *context)
 {
-	struct util_coll_mc *coll_mc = (struct util_coll_mc *) coll_addr;
+	struct util_coll_mc *coll_mc;
 	struct util_coll_comp_item *comp_item;
 	int ret;
 
+	coll_mc = (struct util_coll_mc*) ((uintptr_t) coll_addr);
 	comp_item = calloc(1, sizeof(*comp_item));
 	if (!comp_item)
 		return -FI_ENOMEM;
 
-	comp_item->data = (void *) calloc(1, sizeof(uint32_t));
+	comp_item->data = calloc(1, sizeof(uint32_t));
```

Comment:
Maybe I should have kept reading before posting that last comment.  Maybe data can be a union?  This is a weird calloc, because it's not needed (data can store a u32) and doesn't make it clear why we want a u32 value...

Maybe we do a wholesale replacement of the collective code?